### PR TITLE
IR JS backend

### PR DESCRIPTION
**Pull request by jmfayard**

### CODE REVIEW

**Comment by ajalt** [comment]
Thanks for the contribution. 

1. I'd prefer to enable the IR through the gradle DSL rather than a compiler flag
2. We'd need to update kotest
3. Ultimately, we're blocked from enabling this by [KT-43490](https://youtrack.jetbrains.com/issue/KT-43490). I don't know of a workaround that wouldn't break backwards compatibility.

**Comment by jmfayard** [comment]
Thanks for the information, I opened an issue for kotest here https://github.com/kotest/kotest/issues/2251

**Comment by jmfayard** [comment]
kotest is now compatible with the IR backend!
I upgraded the dependency
and added a comment to document that this issue is correctly blocked by a bug in the IR backend

File: clikt/build.gradle.kts [hunk 1]
```diff
@@ -21,6 +21,14 @@ buildscript {
 
 kotlin {
     jvm()
+
+    /**
```
Comment (LouisCAD) [inline clikt/build.gradle.kts:25]: This shouldn't be a KDoc style comment because it's not a declaration but a function call.
```suggestion
    /*
```

**Pull request merged**